server/neptune/workflows/internal/temporal/selector.go

```diff
@@ -0,0 +1,22 @@
+package temporal
+
+import (
+	"go.temporal.io/sdk/workflow"
+	"time"
+)
+
+// SelectorWithTimeout makes it a bit easier to add Timeout futures
+// by ensuring that we are creating a new cancellable context and returning
+// the cancel func to callers.
+type SelectorWithTimeout struct {
+	workflow.Selector
+}
+
+func (s *SelectorWithTimeout) AddTimeout(ctx workflow.Context, timeout time.Duration, onTimeout func(f workflow.Future)) (workflow.CancelFunc, *SelectorWithTimeout) {
```

Comment:
When reviewing this, I was confused about why we return the Selector, particularly given that every call site ignores that part of the return value. But I realize that's consistent with the other behavior of the interface.

Do you know why the interface is choosing to do that though?